src/Raven.Server/Commercial/LetsEncrypt/SettingsZipFileHelper.cs

```diff
@@ -0,0 +1,582 @@
+﻿using System;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Org.BouncyCastle.Crypto.Prng;
+using Org.BouncyCastle.Pkcs;
+using Org.BouncyCastle.Security;
+using Raven.Client.Documents.Operations;
+using Raven.Client.Exceptions;
+using Raven.Client.Exceptions.Security;
+using Raven.Client.ServerWide.Operations.Certificates;
+using Raven.Server.Config;
+using Raven.Server.Config.Categories;
+using Raven.Server.ServerWide;
+using Raven.Server.Utils;
+using Raven.Server.Utils.Cli;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Sparrow.Platform;
+using Sparrow.Server.Platform.Posix;
+using Sparrow.Threading;
+using Sparrow.Utils;
+using StudioConfiguration = Raven.Client.Documents.Operations.Configuration.StudioConfiguration;
+
+namespace Raven.Server.Commercial.LetsEncrypt;
+
+public static class SettingsZipFileHelper
+{
+    public static async Task<byte[]> CompleteClusterConfigurationAndGetSettingsZip(CompleteClusterConfigurationParameters parameters)
```

Comment:
I wonder if the part "CompleteClusterConfiguration" should be handled by `SettingsZipFileHelper`? IMO that helper should just be responsible for "GetSettingsZip" part. 

Are we able to separate that?

Reply:
Lets consider this:
- Moving `CompleteAuthorizationAndGetCertificate` to `LetsEncryptUtils` it is more suitable.
- Moving `BuildPfx` to `CertificateUtils`
- Changing `WriteSettingsJsonLocally` to `WriteJsonFile` and will be under `LocalWrites` or `WriteToDisk`
I'm not sure myself what's better than keeping it in the way it was...

Reply:
I made changes to the function.
I think it's much more readable and "cleaner"